src/app/container/version-modal/version-modal.component.ts

```diff
@@ -0,0 +1,242 @@
+    if (missingWDL && missingWDL.length > 0) {
+      this.paramfilesService.deleteFiles(this.tool.id, missingWDL, this.version.name, 'WDL').subscribe();
+    }
+    this.containerTagsService.putTags(this.tool.id, this.unsavedVersion).subscribe(response => {
+      this.versionModalService.setIsModalShown(false);
+    });
+  }
+
+  deleteTag() {
+    this.containerTagsService.deleteTag(this.tool.id, this.unsavedVersion.id).subscribe(deleteResponse => {
+      this.containerTagsService.getTags(this.tool.id).subscribe(response => {
+        this.tool.tags = response;
+        this.containerService.setTool(this.tool);
+      });
+    });
+  }
+
+  closeModal() {
+    this.versionModalService.setIsModalShown(false);
+    this.versionModalService.setCurrentMode(null);
+  }
+
+  setMode(mode: TagEditorMode) {
+    this.unsavedCWLTestParameterFilePaths = [];
+    this.unsavedWDLTestParameterFilePaths = [];
+    this.savedCWLTestParameterFilePaths = [];
+    this.savedWDLTestParameterFilePaths = [];
+    this.paramfilesService.getFiles(this.tool.id, 'containers', this.version.name, 'CWL').subscribe(file => {
+      this.savedCWLTestParameterFiles = file;
+      this.savedCWLTestParameterFiles.forEach((fileObject) => {
+        this.savedCWLTestParameterFilePaths.push(fileObject.path);
+      });
+      this.unsavedCWLTestParameterFilePaths = this.savedCWLTestParameterFilePaths.slice();
+
+    });
+    this.paramfilesService.getFiles(this.tool.id, 'containers', this.version.name, 'WDL').subscribe(file => {
+      this.savedWDLTestParameterFiles = file;
+      this.savedWDLTestParameterFiles.forEach((fileObject) => {
+        this.savedWDLTestParameterFilePaths.push(fileObject.path);
+      });
+      this.unsavedWDLTestParameterFilePaths = this.savedWDLTestParameterFilePaths.slice();
+    });
+  }
+
+  addTestParameterFile(descriptorType: DescriptorType) {
+    switch (descriptorType) {
+      case DescriptorType.CWL: {
+        const newTestFile = this.unsavedTestCWLFile;
+        this.unsavedCWLTestParameterFilePaths.push(newTestFile);
+        this.unsavedTestCWLFile = '';
+        break;
+      }
+      case DescriptorType.WDL: {
+        const newTestFile = this.unsavedTestWDLFile;
+        this.unsavedWDLTestParameterFilePaths.push(newTestFile);
+        this.unsavedTestWDLFile = '';
+        break;
+      }
+      default: {
+        console.log('No idea how you submitted in neither edit or add mode');
+      }
+    }
+  }
+
+  removeTestParameterFile(index: number, descriptorType: DescriptorType) {
+    switch (descriptorType) {
+      case DescriptorType.CWL: {
+        this.unsavedCWLTestParameterFilePaths.splice(index, 1);
+        break;
+      }
+      case DescriptorType.WDL: {
+        this.unsavedWDLTestParameterFilePaths.splice(index, 1);
+        break;
+      }
+      default: {
+        console.log('No idea how you submitted in neither edit or add mode');
```

Comment:
This seems repeated multiple times as well, DRY, make this a method and change it in just one place.